portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -36,24 +39,28 @@ public void doGet(HttpServletRequest request, HttpServletResponse response) thro
 
     DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
     
+    // keeping this just in case I need a filter for future reference
     // Filter movieFilter = new FilterPredicate("movie", FilterOperator.EQUAL, movie);
     // Query query = new Query("Comment").setFilter(movieFilter).addSort("timestamp", SortDirection.DESCENDING);
+    
     Query query = new Query(movie).addSort("timestamp", SortDirection.DESCENDING);
     PreparedQuery pq = datastore.prepare(query);
     List<Entity> resultsList = pq.asList(FetchOptions.Builder.withLimit(maxComments));
 
-    ArrayList<String> comments = new ArrayList<>();
+    ArrayList<HashMap<String, String>> comments = new ArrayList<>();
```

Comment:
Could you explain why you need a list of hashmaps here? seems a bit overkill, but maybe i don't understand what you're trying to do (: Seems like each map would just contain a username and a comment? In that case, you can make this a list of pairs.
e.g. List<Pair<String, String>>

Also, use ImmutableList here too :)

Reply:
Oh wow I didn't realize it's not in Java 8! I think a better option would to make a class or enum to store the two strings together, but I won't block merging on it for now (: